src/language/parser.js

```diff
@@ -488,6 +501,17 @@ function parseFragment(
 function parseFragmentDefinition(lexer: Lexer<*>): FragmentDefinitionNode {
   const start = lexer.token;
   expectKeyword(lexer, 'fragment');
+  if (lexer.options.fragmentVariablesExperimental) {
```

Comment:
Add a comment above this that includes the altered grammar with a bit of explanation

```
// Experimental support for defining variables within fragments changes 
// the grammar of FragmentDefinition:
//   - fragment FragmentName on TypeCondition VariableDefinitions? Directives? SelectionSet 
```